File: backend/recpro-backend-lib/src/main/java/de/ubt/ai4/petter/recpro/recpro_backend_lib/recpro/model/recommendation/modeling/dimension/EnvironmentalParameterRecommendationDimension.java
package de.ubt.ai4.petter.recpro.recpro_backend_lib.recpro.model.recommendation.modeling.dimension;

import de.ubt.ai4.petter.recpro.recpro_backend_lib.recpro.model.ercData.environmentalParameter.modeling.AbstractEnvironmentalParameter;
import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.EqualsAndHashCode;
import lombok.NoArgsConstructor;
import lombok.experimental.SuperBuilder;

@SuperBuilder
@AllArgsConstructor
@NoArgsConstructor
@Data
@EqualsAndHashCode(callSuper = true)
@Entity

@DiscriminatorValue("ENVIRONMENTAL_PARAMETER")
public class EnvironmentalParameterRecommendationDimension extends RecommendationDimension {

    @ManyToOne
    private AbstractEnvironmentalParameter environmentalParameter;
}

File: frontend/recpro-frontend-lib/projects/recpro-frontend-lib/src/lib/recommendation/modeling/dimension/model/EnvironmentalParameterRecommendationDimension.ts
import {RecommendationDimension} from './RecommendationDimension';
import {AbstractEnvironmentalParameter, ParameterType} from '../../../../erc-data';

export class EnvironmentalParameterRecommendationDimension extends RecommendationDimension {
  environmentalParameter: AbstractEnvironmentalParameter = new AbstractEnvironmentalParameter();
}
